### ISSUE TITLE
There is no such pid! ->Failed to locate opened application window with appId: C:/Windows/System32/calc.exe, and processId: 2220

### ISSUE THREAD

**Issue by HenningL** [microsoft/WinAppDriver#1334]
When running in Jenkins on a windows slave, this error occurs:
Failed to locate opened application window with appId: C:/Windows/System32/calc.exe, and processId: 2220

The App is up and running but with another pid and the logged pid doesnt even exist.

I read about this issue and all i could find in the inet  : grabbing the window handle for an app with a specific name, but!!! that is not a solution. what if multiple apps of the same run in that windows instance for multithreading purposes.

So my question is , if there is any workaround to identify the app for a specific session. And is there no way to make that pid lookup properly handled by WAD? And if not, why? Just to understand how a workaround might be implemented. Looking up a hardcoded Window-Titel doesnt seem a solution when it comes to a generic approach where the App-Under-Test is not known in compile time.

And last, is there a way to pass something to the app which might be available at runtime.

**Comment by trashbat** [comment]
Would looking up the process name as a child process of winappdriver.exe be a viable workaround?

**Comment by HenningL** [comment]
I dont think so if you have more than one application spinned up by the same WAD instance.
